runtime/lua/vim/ui.lua

```diff
@@ -114,14 +114,19 @@ end
 --- Examples:
 ---
 --- ```lua
+--- -- Asynchronous.
 --- vim.ui.open("https://neovim.io/")
 --- vim.ui.open("~/path/to/file")
---- vim.ui.open("$VIMRUNTIME")
+--- -- Synchronous (wait until the process exits).
+--- local cmd, err = vim.ui.open("$VIMRUNTIME")
+--- if cmd then
+---   cmd:wait()
+--- end
```

Comment:
Maybe we can do better than this pattern. Need to think carefully about https://github.com/neovim/neovim/issues/19624